data/database.py

```diff
@@ -1,58 +1,177 @@
+        """
+        for room in rooms:
             try:
-                room_type = "office" if isinstance(room, Office) \
-                                                        else "livingspace"
+                room_type = room_type_mapping(type(room))
                 c.execute("INSERT INTO room_table (name, type, capacity) \
-                                    VALUES ({}, {}, {})".format(room.name,\
-                                                room_type, room.total_space))
+                          VALUES ('{}', '{}', '{}')".format(
+                                room.name, room_type, room.total_space))
             except sqlite3.IntegrityError:
                 print('ERROR: failed to insert room {}'.format(room.name))
+
+    def insert_people(self, person_list, c):
+        """
+        This function executes the sql command insert to
+        save each person in the Database
+        """
+        for person in person_list:
+            office = person.office.name if person.office is not None else ""
+            c.execute(
+                "INSERT INTO person_table (id, name, designation, " +
+                "office) VALUES ('{}', '{}', '{}', '{}')".format(
+                    person.ID, person.name, person.designation, office))
+            if person.designation.lower() == "fellow":
+                livingSpace = person.livingspace.name if \
+                                person.livingspace is not None else ""
+                c.execute("INSERT INTO livingspace_table (ids, \
+                        wants_accommodation, livingspace) VALUES \
+                            ('{}', '{}', '{}')".format(
+                            person.ID,
+                            person.wants_accommodation, livingSpace))
+
+    def db_exist(self, db_name):
+        """This function check if a database file exist"""
+        return True if os.path.isfile("data/{}.sqlite".format(db_name)) \
+            else False
+
+    def generate_name(self):
+        """
+        This function generate a database name from concatenating
+        the year, month, day, hour, minute and second of the moment
+        """
+        i = datetime.datetime.now()
+        db_name = str(i.year) + str(i.month) + str(i.day) + str(i.hour) \
+            + str(i.minute) + str(i.second)
+        while self.db_exist(db_name):
+            db_name = str(int(db_name) + 1)
+        return db_name
+
+    def run_migrations(self, c):
+        """This function creates table for room, person and livingspace"""
+        c.execute("CREATE TABLE room_table (name TEXT PRIMARY KEY, \
+                                        type TEXT, capacity INTEGER)")
+
+        c.execute("CREATE TABLE person_table (id TEXT PRIMARY KEY,\
+                            name TEXT, designation TEXT, office TEXT)")
+
+        c.execute("CREATE TABLE livingspace_table (ids TEXT PRIMARY KEY,\
+                            wants_accommodation TEXT, livingspace TEXT)")
```

Comment:
why are the created tables `room, person and livingspace`? Since livingspace is here why not also have Office, Staff and Person?